txnkv/transaction/txn.go

```diff
@@ -758,14 +771,33 @@ func (txn *KVTxn) LockKeys(ctx context.Context, lockCtx *tikv.LockCtx, keysInput
 				}
 			}
 		}
+
+		if lockCtx.LockOnlyIfExists && valExists == tikv.SetKeyLockedValueNotExists {
+			skipedLockKeys++
+			continue
+		}
 		memBuf.UpdateFlags(key, tikv.SetKeyLocked, tikv.DelNeedCheckExists, valExists)
 	}
-	txn.lockedCnt += len(keys)
+	txn.lockedCnt += len(keys) - skipedLockKeys
 	return nil
 }
 
+// GetLcokedCount is used for testcase
+func (txn *KVTxn) GetLcokedCount() int {
+	return txn.lockedCnt
+}
+
+// GetPrimaryKey is used for testcase
+func (txn *KVTxn) GetPrimaryKey() []byte {
```

Comment:
If these methods are only used in tests, you can consider add them to `TxnProbe` instead of KVTxn itself. There already exists many useful functions. Your `GetPrimaryKey` can actually be replaced by `txnProbe.GetCommitter().GetPrimaryKey()`

Reply:
done